### PR TITLE
Migrate slice() to new scalar framework

### PR DESCRIPTION
**Pull request by geraint0923**
Migrate array_slice to new framework

### CODE REVIEW

**Comment by geraint0923** [comment]
@erichwang  @cberner

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/ArraySliceFunction.java [hunk 1]
```diff
@@ -13,61 +13,28 @@
...
-        MethodHandle methodHandle = METHOD_HANDLE.bindTo(type);
-        return new ScalarFunctionImplementation(false, ImmutableList.of(false, false, false), methodHandle, isDeterministic());
-    }
-
-    public static Block slice(Type type, Block array, long fromIndex, long length)
+    private ArraySliceFunction() {}
+
+    @TypeParameter("E")
+    @SqlType("array(E)")
+    public static Block slice(@TypeParameter("E") Type type,
```
Comment (haozhun) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/ArraySliceFunction.java]: When function parameters are declared on multiple lines, always put the first one on its own line.

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/ArraySliceFunction.java [hunk 2]
```diff
@@ -13,61 +13,28 @@
...
+            @SqlType("array(E)") Block array,
+            @SqlType(BIGINT) long fromIndex,
```
Comment (haozhun) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/ArraySliceFunction.java]: Generally, we don't statically import `StandardTypes.*`

**Comment by geraint0923** [comment]
Addressed the comments.

**Comment by haozhun** [comment]
Looks good

**Pull request merged**